miasm2/ir/translators/C.py

```diff
@@ -194,38 +284,123 @@ def from_ExprOp(self, expr):
                   expr.op in ["fadd", "fsub", "fdiv", 'fmul', "fscale",
                               "fprem", "fprem_lsb", "fyl2x", "fpatan"]):
-                return "fpu_%s%d(%s, %s)" % (
-                    expr.op,
-                    expr.size,
-                    self.from_expr(expr.args[0]),
-                    self.from_expr(expr.args[1]),
-                )
+                arg0 = self.from_expr(expr.args[0])
+                arg1 = self.from_expr(expr.args[1])
+                if not expr.args[0].size <= self.NATIVE_INT_MAX_SIZE:
+                    arg0 = 'bignum_to_uint64(%s)' % arg0
+                    arg1 = 'bignum_to_uint64(%s)' % arg1
+                out = "fpu_%s%d(%s, %s)" % (expr.op, expr.size, arg0, arg1)
+                return out
+
             elif expr.op == "segm":
                 return "segm2addr(jitcpu, %s, %s)" % (
                     self.from_expr(expr.args[0]),
                     self.from_expr(expr.args[1])
                 )
-            elif expr.op in ['udiv', 'umod', 'idiv', 'imod']:
-                return '%s%d(%s, %s)' % (
-                    expr.op,
-                    expr.args[0].size,
-                    self.from_expr(expr.args[0]),
-                    self.from_expr(expr.args[1])
-                )
+
+            elif expr.op in ['udiv', 'umod']:
+                arg0 = self.from_expr(expr.args[0])
+                arg1 = self.from_expr(expr.args[1])
+
+                if expr.size <= self.NATIVE_INT_MAX_SIZE:
+                    out = '%s%d(%s, %s)' % (
+                        expr.op,
+                        expr.args[0].size,
+                        self.from_expr(expr.args[0]),
+                        self.from_expr(expr.args[1])
+                    )
+                else:
+                    out = "bignum_%s(%s, %s)" % (
+                        expr.op,
+                        self.from_expr(expr.args[0]),
+                        self.from_expr(expr.args[1])
+                    )
+                    out = "bignum_mask(%s, %d)"% (out, expr.size)
+                return out
+
+
+
+            elif expr.op in ['idiv', 'imod']:
+                arg0 = self.from_expr(expr.args[0])
+                arg1 = self.from_expr(expr.args[1])
+
+                if expr.size <= self.NATIVE_INT_MAX_SIZE:
+                    out = '%s%d(%s, %s)' % (
+                        expr.op,
+                        expr.args[0].size,
+                        self.from_expr(expr.args[0]),
+                        self.from_expr(expr.args[1])
+                    )
+                else:
+                    out = "bignum_%s(%s, %s, %d)" % (
+                        expr.op,
+                        self.from_expr(expr.args[0]),
+                        self.from_expr(expr.args[1]),
+                        expr.size
+                    )
+                    out = "bignum_mask(%s, %d)"% (out, expr.size)
+                return out
+
+            elif expr.op == "/":
```

Comment:
I'm not sure "/" and "%" are still supported, because there are ambiguous compared to "udiv/idiv" and "umod/imod"

Reply:
Done
I completely remove those operators in the translator:
There were a last use of those operators in the x86 `aam`.
I added the UDIV8/UMOD8 in the op_semantic which where missing and needed to update `aam`